src/array/sortBy.ts

```diff
@@ -16,8 +16,22 @@ declare global {
   }
 }
 
-Array.prototype.sortBy = function <T, K extends Comparable>(this: T[], fn: (el: T) => K): T[] {
-  return [...this].sort((a, b) => compare(fn(a), fn(b)));
+Array.prototype.sortBy = function <T, K extends Comparable>(this: T[], fn: (el: T) => K | K[]): T[] {
+  return [...this].sort((a, b) => {
+    const av = fn(a);
+    const bv = fn(b);
+    if (Array.isArray(av) && Array.isArray(bv)) {
+      for (let i = 0; i < av.length; i++) {
```

Comment:
issue: i think we need to consider if the arrays have different lengths? eg if `bv` was `[1,2,3]` elements and `av` was `[1,2]`, then `bv` should sort before, but with this code they'd sort as equal.

Reply:
I was thinking of finding some fancy TypeScript that would force the lambda to always return the same length-d array; ChatGTP had some suggestions:

```ts
type LambdaWithFixedLength<N extends number, T = unknown> = (...args: any[]) => FixedLengthArray<T, N>;

type FixedLengthArray<T, N extends number, R extends unknown[] = []> = R["length"] extends N
  ? R
  : FixedLengthArray<T, N, [T, ...R]>;
```

But I couldn't get it to work in the time box I had for adding this, so was pretty tempted to be satisfied with the "don't do that" version of the lambda returning different length-d arrays.

I suppose could do a runtime error if they're different lengths. Pushed that up.

Does that sgty?

Reply:
I dunno i feel like different length arrays is a valid use case and not that hard to support? eg, lets say you were doing semantic versioning and to sort you emitted the version as an array of numbers

Reply:
:thinking: maybe I guess; personally I'd assume that different length arrays in the wild would 80% of the time be bugs the programmer didn't know/mean to write. :cry:

Reply:
i think you're just envisioning the use case where you'd have like `foos.sortBy((foo) => [foo.bar, foo.baz])` and i'm envisioning something more dynamic? I don't think the dynamic case is that unrealistic to use. Also it'd be like a 1 line change: 

```suggestion
      const length = Math.max(av.length, bv.length);
      for (let i = 0; i < length; i++) {
```

Reply:
Well, 6 months later :sweat_smile: , I want to use this; still little "want to see the real-world use case" before supporting dynamic lengths, but also just want to hit merge to unblock my other work.